### PR TITLE
[Core]Fix coredump when m_pCryptoControl is Null

### PR DESCRIPTION
**Pull request by kyolong**
#1182 
maybe fixed ~

### CODE REVIEW

File: srtcore/core.cpp [hunk 1]
```diff
@@ -8906,7 +8906,11 @@ std::pair<int, steady_clock::time_point> CUDT::packData(CPacket& w_packet)
             // together with encrypting, and the packet should be sent as is, when rexmitting.
             // It would be nice to research as to whether CSndBuffer::Block::m_iMsgNoBitset field
             // isn't a useless redundant state copy. If it is, then taking the flags here can be removed.
-            kflg    = m_pCryptoControl->getSndCryptoFlags();
+            if(m_pCryptoControl)
```
Comment (maxsharabayko) [inline srtcore/core.cpp]:
```suggestion
            if (m_pCryptoControl)            
```

**Comment by ethouris** [comment]
I'd keenly see the debug log - it will be cleanly seen in what order things happen and therefore how it was possible that `packData` was called at all on a socket being closed. AFAIR there should be first the `m_bClosing` and `m_bBroken` flags set first, then the crypto object deleted. And every function that would do any next thing with the data should first check these closing/broken flags before doing anything.

**Comment by ethouris** [comment]
My suggestion:

For testing, keep this change, but in the condition when `!m_pCryptoControl` add a log message so that you can see that it has happened, but print also the state. Something like:
```
LOGC(mglog.Error, log << "CryptoControl DELETED. connected=" << m_bConnected << " broken=" << m_bBroken << " opened=" << m_bOpened);
```
When you can see this log during testing, please see if when it happens the broken flag is set and possibly also connected flag is clear. If it is so, it means that the code works still correctly predictably, so the proper fix should be:
```
if (m_bBroken || !m_bConnected)
{
       return std::make_pair(0, enter_time);
}
```
in the beginning of `packData` method (at least after `enter_time` was assigned).

**Comment by maxsharabayko** [comment]
Closing as fixed by #1315

**Pull request closed without merging**